CMakeLists.txt

```diff
@@ -116,6 +116,12 @@ if(BUILD_WITH_SANITIZERS AND NOT WIN32)
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -fno-omit-frame-pointer -fsanitize=${SANITIZERS}")
 endif()
 
+if(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
```

Comment:
Would be good if this were controlled by an option to avoid warnings when compilers change and start complaining about things that today's compilers do not complain about.  Or in environments where there are warnings one wants to ignore.

Reply:
Will add option but have it default on for now.